docs/tutorials/data_provider.md

```diff
@@ -61,3 +61,100 @@ impl AdditiveIdentity {
     }
 }
 ```
+
+## Caching Data Provider
+
+It is not generally required to implement a cache, especially if using the BakedDataProvider. However, if you wish to implement a cache in your data pipeline, you can do so like this:
```

Comment:
Elaborate on this? Why is it not suggested for baked? In what cases is it suggested.

Reply:
I added more explanatory prose.